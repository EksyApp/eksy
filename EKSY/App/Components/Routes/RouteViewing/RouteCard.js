import React, {Component} from 'react'
import Card from '../../Common/Card'
import {TouchableWithoutFeedback, StyleSheet} from 'react-native'
import CardSection from '../../Common/CardSection'
import Label from '../../Common/Label'
import TextArea from '../../Common/TextArea'
import PropTypes from 'prop-types'
import {RouteShape, StyleShape} from "../../../Utils/PropTypeShapes";

export default class RouteCard extends Component {
	
	render() {
		return(
				<TouchableWithoutFeedback onPress={() => this.props.onPress(this.props.route)}>
					<Card style={[styles.container, {width: this.props.width}, this.props.style]}>
						<CardSection style={styles.infoContainer}>
							<Label style={styles.title}>{this.props.route.title}</Label>
						</CardSection>
						
						<TextArea>{this.props.route.text}</TextArea>
					</Card>
				</TouchableWithoutFeedback>
		)
	}
	
}

const styles = StyleSheet.create({
	container: {
		height: 200,
		marginLeft: 0,
		width: '100%'
	},
	
	infoContainer: {
		width: "100%",
		alignItems: "center",
		justifyContent: "center",
		overflow: "scroll"
	},
	
	title: {
		marginTop: 10
	}
});

RouteCard.propTypes = {
	onPress: PropTypes.func,
	route: RouteShape,
	width: PropTypes.number,
	style: StyleShape,
}