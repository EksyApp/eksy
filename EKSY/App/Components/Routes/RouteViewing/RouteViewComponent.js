import React, {Component} from 'react'
import {View, StyleSheet, ScrollView} from 'react-native'
import Header from '../../Common/Header'
import * as Theme from '../../../Theme/index'
import RouteView from '../../Common/RouteView'
import Button from '../../Common/Button'
import PropTypes from 'prop-types'
import {RegionShape, RouteShape} from "../../../Utils/PropTypeShapes";
import Container from '../../Common/Container'

//Renders a view that contains RouteView and edit button
//RouteView shows the route's metadata
//Edit button takes to EditRouteComponent
export default class RouteViewComponent extends Component {

	render() {
		return(
				<Container backButton title="Route">
					<RouteView
							initialRegion={this.props.currentRegion}
							route={this.props.route}
							onMarkerClick={this.props.onMarkerClick}
					/>
					<Button onPress={() => this.props.onEditClick()}>
						Edit
					</Button>
				</Container>
		)
	}

}

const styles = StyleSheet.create({
	container: {
		flex: 1,
		backgroundColor: Theme.backgroundColor
	},

})

RouteViewComponent.propTypes = {
	currentRegion: RegionShape,
	route: RouteShape,
	onMarkerClick: PropTypes.func
}
