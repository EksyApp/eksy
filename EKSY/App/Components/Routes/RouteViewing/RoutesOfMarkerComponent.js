import React, {Component} from 'react'
import Container from '../../Common/Container'
import RouteCardList from './RouteCardList'
import {StyleSheet} from 'react-native'
import Message from '../../Common/Message'

export default class RoutesOfMarkerComponent extends Component {
	
	renderList() {
		if(this.props.loading) {
			return(<Message>Loading...</Message>)
		} else {
		
		}
	}
	
	render() {
		return(
				<Container backButton title="Start a route">
					<RouteCardList routes={this.props.routes} onPress={(route) => this.props.onCardClick(route)} style={styles.card}/>
				</Container>
		)
	}
	
}

const styles = StyleSheet.create({
	card: {
		width: '95%',
		alignSelf: 'center'
	}
})
