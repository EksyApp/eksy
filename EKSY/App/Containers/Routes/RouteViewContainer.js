import React, {Component} from 'react'
import {connect} from 'react-redux'
import RouteViewComponent from '../../Components/Routes/RouteViewComponent'
import {Actions} from 'react-native-router-flux'
import * as ReduxActions from '../../Actions'

export class RouteViewContainer extends Component {
	
	handleMarkerClick(marker) {
		this.props.setMarkerSelected(marker)
		Actions.markerView()
	}
	
	render() {
		return(
				<RouteViewComponent
						route={this.props.route}
						currentRegion={this.props.currentRegion}
						onMarkerClick={(marker) => this.handleMarkerClick(marker)}
				/>
		)
	}
	
	
}

const mapStateToProps = (state) => {
	return {
		route: state.routes.routeSelected,
		currentRegion: state.map.currentRegion
	}
}

const mapDispatchToProps = (dispatch) => {
	return {
		setMarkerSelected: (marker) => {dispatch(ReduxActions.setMarkerSelected(marker))}
	}
}

export default connect(mapStateToProps, mapDispatchToProps)(RouteViewContainer)