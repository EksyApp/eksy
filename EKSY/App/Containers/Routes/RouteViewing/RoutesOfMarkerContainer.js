import React, {Component} from 'react'
import RoutesOfMarkerComponent from '../../../Components/Routes/RouteViewing/RoutesOfMarkerComponent'
import {connect} from 'react-redux'
import * as ReduxActions from '../../../Actions/'
import Dao from '../../../Dao/Dao'
import {Actions} from 'react-native-router-flux'

export class RoutesOfMarkerContainer extends Component {
	
	constructor(props) {
		super(props)
		
		this.state={
			loading: true,
			routes: []
		}
		this.getRoutes()
		
	}
	
	refresh() {
		this.setState({loading: true})
		this.getRoutes()
	}
	
	async getRoutes() {
		let routes = await new Dao().getRoutesOfMarker(this.props.marker)
		routes = routes.filter((route) => route.markers.length >= 2)
		this.setState({routes, loading:false})
	}
	
	handleCardClick(route) {
		if(route.markers.length >= 2) {
			this.props.setActiveRoute(route)
			this.props.setNextMarker(route.markers[1])
			this.props.routeIsActive(true)
			Actions.pop()
		}
	}
	
	render() {
		return(
				<RoutesOfMarkerComponent
						routes={this.state.routes}
						loading={this.state.loading}
						onCardClick={(route) => this.handleCardClick(route)}
				/>
		)
	}
	
}

const mapStateToProps = (state) => {
	return {
		marker: state.markers.markerSelected,
	}
}

const mapDispatchToProps = (dispatch) => {
	return {
		setActiveRoute: (route) => {dispatch(ReduxActions.setRouteActive(route))},
		routeIsActive: (state) => {dispatch(ReduxActions.routeIsActive(state))},
		setNextMarker: (marker) => {dispatch(ReduxActions.setNextMarker(marker))}
	}
}

export default connect(mapStateToProps, mapDispatchToProps)(RoutesOfMarkerContainer)