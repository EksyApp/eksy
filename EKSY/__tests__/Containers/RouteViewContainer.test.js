import React from 'react'
import { RouteViewContainer } from '../../App/Containers/Routes/RouteViewing/RouteViewContainer'
import renderer from 'react-test-renderer'

test('RouteViewContainer renders correctly', () => {
	const tree = renderer.create(
			<RouteViewContainer
					setMarkerSelected={() => {}}
					route={{title: "", text: "", markers: [{latitude: 1, longitude: 1, title: "", text: ""}]}}
					currentRegion={{latitude: 1, longitude: 1}}
			/>
	).toJSON();
	expect(tree).toMatchSnapshot();
});