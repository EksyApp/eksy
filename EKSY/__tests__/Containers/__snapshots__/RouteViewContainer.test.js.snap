// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RouteViewContainer renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgb(243,244,229)",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "rgb(226,227,217)",
        "height": 50,
        "justifyContent": "center",
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Array [
            false,
            false,
            false,
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "justifyContent": "center",
            },
            Object {
              "height": 50,
              "left": 0,
              "position": "absolute",
              "top": 0,
              "width": 50,
              "zIndex": 1,
            },
          ],
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "black",
              "fontSize": 35,
            },
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              Object {
                "color": "rgb(101,114,122)",
              },
            ],
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        î—„
      </Text>
    </View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "rgb(101,114,122)",
            "fontSize": 20,
            "fontWeight": "bold",
            "textAlign": "center",
          },
          undefined,
        ]
      }
    >
      Route
    </Text>
  </View>
  <RCTScrollView
    backButton={true}
    style={
      Array [
        Object {
          "backgroundColor": "rgb(243,244,229)",
          "flex": 1,
        },
        undefined,
      ]
    }
    title="Route"
  >
    <View>
      <View>
        <View
          style={
            Object {
              "height": 300,
              "marginBottom": 20,
              "width": "100%",
            }
          }
        >
          <MapView
            initialRegion={
              Object {
                "latitude": 1,
                "longitude": 1,
              }
            }
            style={
              Object {
                "height": "100%",
                "width": "100%",
              }
            }
          >
            <Marker
              coordinate={
                Object {
                  "latitude": 1,
                  "longitude": 1,
                }
              }
            />
            <Polyline
              coordinates={
                Array [
                  Object {
                    "latitude": 1,
                    "longitude": 1,
                  },
                ]
              }
            />
          </MapView>
        </View>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "rgb(101,114,122)",
                "fontSize": 20,
                "fontWeight": "bold",
                "textAlign": "center",
              },
              undefined,
            ]
          }
        >
          
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "rgb(89,85,78)",
                "fontSize": 18,
                "margin": 10,
              },
              undefined,
            ]
          }
        >
          
        </Text>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#e1e8ee",
                "height": 1,
              },
              Array [
                Object {
                  "backgroundColor": "rgb(112,112,112)",
                  "height": 1,
                  "margin": 3,
                  "marginLeft": 10,
                  "marginRight": 10,
                },
                undefined,
              ],
            ]
          }
        />
        <RCTScrollView
          dataSource={
            ListViewDataSource {
              "items": 1,
            }
          }
          enableEmptySections={true}
          renderRow={[Function]}
          renderScrollComponent={[Function]}
        >
          <View>
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              hitSlop={undefined}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "backgroundColor": "rgb(226,227,217)",
                    "borderBottomWidth": 0,
                    "borderColor": "#ddd",
                    "borderRadius": 2,
                    "borderWidth": 1,
                    "elevation": 1,
                    "marginLeft": 5,
                    "marginTop": 10,
                    "shadowColor": "#000",
                    "shadowOffset": Object {
                      "height": 2,
                      "width": 0,
                    },
                    "shadowOpacity": 0.1,
                    "shadowRadius": 2,
                  },
                  Array [
                    Object {
                      "height": 200,
                      "marginLeft": 0,
                      "width": "100%",
                    },
                    Object {
                      "width": 740,
                    },
                    Object {
                      "alignSelf": "center",
                      "width": "95%",
                    },
                  ],
                ]
              }
              testID={undefined}
            >
              <View
                style={
                  Array [
                    Object {
                      "borderBottomWidth": 1,
                      "borderColor": "rgb(112,112,112)",
                      "flexDirection": "row",
                      "justifyContent": "flex-start",
                      "padding": 5,
                      "position": "relative",
                    },
                    Object {
                      "alignItems": "center",
                      "justifyContent": "center",
                      "overflow": "scroll",
                      "width": "100%",
                    },
                  ]
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "rgb(101,114,122)",
                        "fontSize": 20,
                        "fontWeight": "bold",
                        "textAlign": "center",
                      },
                      Object {
                        "marginTop": 10,
                      },
                    ]
                  }
                >
                  
                </Text>
              </View>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": "rgb(89,85,78)",
                      "fontSize": 18,
                      "margin": 10,
                    },
                    undefined,
                  ]
                }
              >
                
              </Text>
            </View>
          </View>
        </RCTScrollView>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignSelf": "stretch",
            "backgroundColor": "rgb(230,232,227)",
            "borderColor": "rgb(101,114,122)",
            "borderRadius": 5,
            "borderWidth": 1,
            "height": 40,
            "marginBottom": 15,
            "marginLeft": 5,
            "marginRight": 5,
            "marginTop": 5,
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "alignSelf": "center",
              "color": "rgb(101,114,122)",
              "fontSize": 16,
              "fontWeight": "600",
              "paddingBottom": 10,
              "paddingTop": 10,
            }
          }
        >
          Edit
        </Text>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
